### ISSUE TITLE
Difference between mobilenet and moilbenet quantisized weights

### ISSUE THREAD

**Issue by foromer4** [ARM-software/ComputeLibrary#637]
In the `[mobilenet example](https://github.com/ARM-software/ComputeLibrary/blob/b9abeae0897bef74553ba9800c4ff5f74131c5b4/examples/graph_mobilenet.cpp) , why is there one path for regular model wrights and another for quantisized? If I understood correctly, the weights should be the same and only in the acl  covert code, does the Quantization info map them from float to uint8.

**Comment by Wheest** [comment]
Your intuitions are close.

Quantisation does use the same weights, at first.  But it is a retraining process.  You use the full float weights as initialisation, and then retrain from there, reducing the bits.  

So you end up with different models, a quantised and non-quantised one.  I don't think ACL handles retraining, this is something that is done on your host framework.

Perhaps the [following paper](https://arxiv.org/abs/1712.05877) will be useful.

**Comment by ManubARM** [comment]
Hi @foromer4 
The networks have few differences. On top of having to handle quantization information in the "get_dwsc_node_qasymm" function there is no BatchNormalizationLayer.

@Wheest is right, we don't have any sort of training/retraining functionality.

Best,
Manuel

**Comment by foromer4** [comment]
@Wheest , @ManubARM  thanks for the input. what I'm interested in doing is **post** quanitzation on a regular trained mobilenet ssd model. In that case, can't I load the regular weights and use the qunatization info which I cac calculate myself for each layer? Must I pre-train a quantisized model before converting it into ACL? thank you

**Comment by ManubARM** [comment]
Yes, you can convert the weights from QASYMM8 to floating point runtime and then give them as F32 tensors to the layers. 

I don't think it is necessary to pre-train the model if the quantization info are carefully chosen.
But you might want to try anyway and evaluate the best model for your case in terms of memory/performance/accuracy.

Hope it helps,
Manuel

**Comment by ManubARM** [comment]
Closing the issue, feel free to reopen otherwise
